### ISSUE TITLE
Receiving - Error: SSL peer certificate or SSH remote key was not OK - after Terraform deployment

### ISSUE THREAD

**Issue by jblewnormal** [ceramicnetwork/terraform-aws-ceramic#19]
I've been able to run the Terraform build successfully and it deploys all resources to my AWS account, but when I ping the ALB's  health check endpoint I continue to receive this error:

`Error: SSL peer certificate or SSH remote key was not OK`

This is the endpoint I'm calling: `ceramic-staging--node-785765069.us-east-2.elb.amazonaws.com/api/v0/node/healthcheck`

All the resources instructed to create before deployment have been created.

The only thing I can think that's causing this is something to do with the ACM certificate and/or domain. I own `normalapi.com` and created a certificate for `idk.normalapi.com` using ACM to generate the ARN shown below. The certificate is valid and setup correctly. This is the domain I pass for the `ipfs_domain_name` variable, but I'm really unsure if this is correct.

The documentation here is incredibly hard to follow and has nothing for troubleshooting so I'm posting all this after hours at this.

Here is my current `variables.tf` where I've set all the values to use as defaults:

```
variable "acm_certificate_arn" {
  type        = string
  description = "ARN of ACM SSL certificate"
  default = "arn:aws:acm:us-east-2:270757412251:certificate/a2014f69-ff65-4b62-a3de-a70fc6bf7a56"
}

variable "aws_region" {
  type        = string
  description = "AWS region. Must match region of vpc_id and public_subnet_ids."
  default = "us-east-2"
}

variable "base_namespace" {
  type        = string
  description = "Base namespace"
  default = ""
}

variable "default_tags" {
  type        = map(any)
  description = "Tags"
  default = {}
}

variable "ecs_cluster_name" {
  type        = string
  description = "Name of ECS cluster"
  default = "CeramicServices"
}

variable "env" {
  type        = string
  description = "Environment name used for namespacing"
  default = ""
}

variable "image_tag" {
  type        = string
  description = "Image tag"
  default = "latest"
}

variable "private_subnet_ids" {
  type        = list(string)
  description = "List of private subnet ids for the VPC"
  default = ["subnet-0127278d26ffbabb0", "subnet-0e32deefde7511bc7"]
}

variable "public_subnet_ids" {
  type        = list(string)
  description = "List of ALB subnet ids"
  default = ["subnet-09a4d736970e2c9b9", "subnet-077bdccc87f440235"]
}

variable "s3_bucket_name" {
  type        = string
  description = "Name of S3 bucket to use as a backend for Ceramic and IPFS"
  default = "ceramicdatastore"
}

variable "vpc_security_group_id" {
  type        = string
  description = "VPC security group id"
  default = "sg-0ca17e66973d1ab13"
}

variable "vpc_id" {
  type        = string
  description = "Id of VPC"
  default = "vpc-01d5cc98c76743628"
}

variable "vpc_cidr_block" {
  type        = string
  description = "Default CIDR block of the VPC"
  default = "10.2.0.0/16"
}

/* Ceramic */

variable "ceramic_anchor_service_api_url" {
  type        = string
  description = "URL for Ceramic Anchor Service API"
  default = "https://mainnet.infura.io/v3/<apiKey>" 
}

variable "ceramic_cors_allowed_origins" {
  type        = string
  description = "Web browser CORS allowed origins as stringified regex"
  default     = ".*"
}

variable "ceramic_cpu" {
  type        = number
  description = "vCPU units to allocate to the Ceramic daemon ECS task"
  default     = 1024 # 1024 = 1 vCPU
}

variable "ceramic_env" {
  type        = string
  description = "Optional. Environment namespace for Ceramic"
  default = "staging"
}

variable "ceramic_efs_logs_fs_id" {
  type        = string
  description = "ID of EFS volume for Ceramic logs"
  default = "fs-08e8ebc50015e2572"
}

variable "ceramic_efs_logs_fs_name" {
  type        = string
  description = "Name of EFS volume for Ceramic logs"
  default = "CeramicLogs"
}

variable "ceramic_eth_rpc_url" {
  type        = string
  description = "Ethereum RPC URL. Must match anchor service ETH network"
  default = "https://mainnet.infura.io/v3/<apiKey>"
}

variable "ceramic_memory" {
  type        = number
  description = "Memory allocation per Ceramic daemon ECS task"
  default     = 2048
}

variable "ceramic_network" {
  type        = string
  description = "Ceramic network (e.g. testnet-clay)"
  default = "testnet-clay"
}

variable "ceramic_task_count" {
  type        = number
  description = "Number of Ceramic ECS tasks to run in the ECS service"
  default     = 1
}

/* IPFS */

variable "ipfs_cpu" {
  type        = number
  description = "vCPU units to allocate to the IPFS ECS task"
  default     = 1024 # 1024 = 1 vCPU
}

variable "ipfs_default_log_level" {
  type        = string
  description = "IPFS default log level"
  default     = "info"
}

variable "ipfs_domain_name" {
  type        = string
  description = "Domain name, including TLD"
  default = "idk.normalapi.com"
}

variable "ipfs_enable_alb_logging" {
  type        = bool
  description = "True to enable ALB logs (stored in a new S3 bucket)"
  default     = false
}

variable "ipfs_enable_repo_backup_to_s3" {
  type        = bool
  description = "True to backup IPFS repo to S3"
  default     = false
}

variable "ipfs_memory" {
  type        = number
  description = "Memory allocation per IPFS API instance"
  default     = 2048
}

variable "ipfs_task_count" {
  type        = number
  description = "Number of IPFS ECS tasks to run in the ECS service"
  default     = 1
}

variable "ipfs_s3_repo_backup_bucket_arn" {
  type        = string
  description = "ARN of S3 bucket to use as a backup for the IPFS repo"
  default     = "arn:aws:s3:::ceramicdatastore"
}

variable "use_existing_ipfs_peer_identity" {
  type        = string
  description = "Use existing IPFS peer identity"
  default     = false
}
```

Let me know if posting any additional code/files would be helpful and I can add them.

Thanks!

Josh

**Comment by 3benbox** [comment]
Hi, it looks like the cert is for the domain name `idk.normalapi.com`, is that what you're using to connect, or the LB name, `ceramic-staging--node-785765069.us-east-2.elb.amazonaws.com`? 

I get a name mismatch on requests for the LB name from the internet.

**Comment by jblewnormal** [comment]
> There might be a mismatch because I've redeployed a couple times over the past few days continuing to test this with different params so the LB endpoint may have changed.

Confirmed that this is NOT the case, the LB endpoint is the exact same, so I'm this naming mismatch could lead to the source. Would you mind sharing it here?

**Comment by 3benbox** [comment]
> idk.normalapi.com always returns Error: Couldn't resolve host name

That's a DNS error that needs to be resolved. The terraform doesn't create the DNS record as it's dependent on your DNS provider. Creating it manually and using your hostname should resolve these 2 errors.

>  Is the ACM cert only used for the LB? 
Yes

> Does the ipfs_domain_name need to be an actual IPFS gateway for connection or is this just for internal connectivity amongst the VPC and ECS? 

From reviewing the terraform, it looks to just be used for internal connectivity amongst the VPC and ECS and announcing the endpoint to peers.

**Comment by jblewnormal** [comment]
Thanks for clarifying those points for me - that really helps. I got the DNS issue resolved so `https://idk.normalapi.com/api/v0/node/healthcheck` is now working but returns a `503` error.

Some possible causes I'm investigating:

1) The IPFS ECS node continues to throw these two errors:
```net/identify identify/id.go:377 failed negotiate identify protocol with peer {"peer": "12D3KooWFpaDunPLy41Fo2Uu3w53apX34agWwvbERh2bvDQe2rQ9", "error": "Application error 0x0: conn-24891382: system: cannot reserve inbound connection: resource limit exceeded"}```

^I've tried doubling the default memory allocation `ipfs_memory` to 4096 but this hasn't seemed to fix anything.

```Error: lock /data/ipfs/repo.lock: someone else has the lock```

2) The Ceramic ECS node continues to move between Pending and Failed, but no logs are being output so I'm working on grabbing these. Will update...

**Comment by jblewnormal** [comment]
Update:

The IPFS ECS node is running correctly, and after fixing a couple issues with the Ceramic node it's getting caught up when trying to connect to the IPFS node.

```FetchError: request to https://go-ipfs-ceramic-staging--internal.idk.normalapi.com:5011/api/v0/id failed, reason: Hostname/IP does not match certificate's altnames: Host: go-ipfs-ceramic-staging--internal.idk.normalapi.com. is not in the cert's altnames: DNS:*.normalapi.com```

`go-ipfs-ceramic-staging--internal.idk` is setup correctly with a CNAME in my domain's DNS records pointing to the internal IPFS LB's endpoint: `internal-ceramic-staging--ipfs-nd-in-558386843.us-east-2.elb.amazonaws.com`.

After seeing this error, I tried switching my ACM cert from `idk.normalapi.com` to `*.normalapi.com` as a catch-all attempt given what the error says, but after redeploying with the new cert ARN the same error is happening (above).

Have you ever seen this before?

**Comment by jblewnormal** [comment]
Update:

I got it working! Had to recreate the `idk.normalapi.com` ACM cert but include the `go-ipfs-ceramic-staging--internal.idk.normalapi.com` as an additional value.

Health check coming back Alive!

Thank you again for your help with this @3benbox - I'm gonna write up some edits and additions I think would be useful to include in the docs/readme of this repo for others in the future so they don't have these same issues. There's a couple holes and confusing descriptions in the docs that led to these missteps that clearing up should make spinning up nodes much easier.
